### ISSUE TITLE
Area2D nodes do not register mouse events on the top or left edges

### ISSUE THREAD

**Issue by skoovo** [godotengine/godot#25462]
**Godot version:** 3.1 BETA 3 (also present in BETA 2)

**OS/device including version:** Windows 10

**Issue description:**
When arranging Area2D nodes into a perfect grid, there are sometimes gaps in coverage where two nodes meet. Clicking along certain edges sometimes results in a spot that doesn't trigger any events on either node. I would expect one of the nodes to detect the click since there is no gap between them.

**Steps to reproduce:**
Create a grid of Area2D nodes and click along the edges of nodes. This doesn't happen along every edge (although that may be due to mouse/DPI alignment), so it may take a moment to find one, but once you have, you'll notice that it runs the length (or width) of the grid.

**Minimal reproduction project:**

I've added a demo project that has this issue. The green buttons on the right are TextureButtons which work as expected. The blue buttons on the left are Area2D nodes that have dead spots. Using this project I have found 4 gaps at the following edges:

Vertical gaps:
X = 114 and X = 306

Horizontal gaps:
Y = 114 and Y= 306

I suspect there may be more edges that experience this, but due to mouse/DPI settings I cannot perfectly align the cursor with other edges. Although this seems like a rare issue, it pops up quite a lot when quickly clicking on buttons and you accidentally hit a boundary resulting in nothing happening.
[Area2DTest.zip](https://github.com/godotengine/godot/files/2807976/Area2DTest.zip)

If you need any other information, please let me know. Thanks!

**Comment by GrayJacketStudios** [comment]
Can confirm it happening in the last version (custom build from the last godot version) I will try to see if I can fix it

**Comment by skoovo** [comment]
I don't know if it is relevant, but it did occur to me that the positions I'm finding this bug in are the final pixel of an area, not the first of the next area.

It makes me wonder if this may be down to how the areas are calculated and are not inclusive of the final pixel. I don't know how Area2D works under the hood, but it has the hallmarks of a < where it needs to be <=? Or perhaps rounding problems if there are floats involved at some level.

I don't I know C++,  but I may dig around myself to try and see if anything obvious jumps out at me.

**Comment by skoovo** [comment]
I tried replacing the RectangleShape2D with ConvexPolygonShape2D in the test project, and the dead spots have disappeared. However, now clicks along boundaries activate BOTH areas. This would seem to indicate some kind of overlap I guess?

For reference my polygon points are:

-32, -32
32, -32
32, 32
-32, 32

**Comment by skoovo** [comment]
The dead spots also occur using the CircleShape2D with a radius of 32.

**Comment by skoovo** [comment]
Just updated to Godot 3.1 BETA 4 and the issue is still present.

Also, I created a different type of test project that collides two of the buttons. The buttons seem to perfectly detect a collision with each other, just not with the mouse input. Maybe these are calculated differently? I have also tested overlapping Area2D nodes to check if  anything strange was happening there, but it seems fine, with both Area2D nodes registering the click.
